crates/wasmtime/src/runtime/vm/cow.rs

```diff
@@ -563,29 +565,25 @@ impl MemoryImageSlot {
         pagemap: Option<&PageMap>,
         keep_resident: HostAlignedByteCount,
         decommit: impl FnMut(*mut u8, usize),
-    ) {
+    ) -> usize {
         assert_eq!(
             vm::decommit_behavior(),
             DecommitBehavior::RestoreOriginalMapping
         );
 
         unsafe {
-            match &self.image {
+            return match &self.image {
```

Comment:
Looks like unnecessary `return` and `;` at end of this match block - the inner change should have been sufficient?